### ISSUE TITLE
Ignore links to #top when checking anchors

### ISSUE THREAD

**Issue by Lucidiot** [Keats/gutenberg#2495]
# Bug Report

## Environment

Zola version: 0.18.0

## Expected Behavior

[The HTML spec](https://html.spec.whatwg.org/#select-the-indicated-part) includes some special handling for a `#top` fragment added to the URL:

> 10. If *decodedFragment* is an [ASCII case-insensitive](https://infra.spec.whatwg.org/#ascii-case-insensitive) match for the string `top`, then return the [top of the document](https://html.spec.whatwg.org/#top-of-the-document).

This means that adding `[some text](#top)` to a page correctly creates a link to the top of the page, and Zola should treat the link as always being valid since the anchor is technically always there.

## Current Behavior

```
$ zola check
Checking site...
Checking all internal links with anchors.
> Checked 1 internal link(s) with anchors: 1 target(s) missing.
Failed to check the site
Error: The anchor in the link `@/index.md#top` in /tmp/testsite/content/index.md does not exist.
```

## Steps to reproduce

Add a Markdown link to `#top` in any page, and run `zola check`.

## Workaround

Linking to `#` can work instead, as the HTML spec also defines it as an edge case:

> 3. If *fragment* is the empty string, then return the special value **top of the document**.

**Comment by Keats** [comment]
I didn't know that!

**Comment by ta-nishq** [comment]
Fix #2519
